lib/flipper/adapters/redis.rb

```diff
@@ -26,15 +26,33 @@ def features
         read_feature_keys
       end
 
+      def self.redis_sadd_returns_boolean?
+        return @redis_sadd_returns_boolean if defined?(@redis_sadd_returns_boolean)
+        @redis_sadd_returns_boolean = true
+          if ::Redis.respond_to?(:sadd_returns_boolean)
+            ::Redis.sadd_returns_boolean
+          else
+            false
+          end
+      end
+
       # Public: Adds a feature to the set of known features.
       def add(feature)
-        @client.sadd FeaturesKey, feature.key
+        if self.class.redis_sadd_returns_boolean?
```

Comment:
Not sure how you might want to address this so the conditional is evaluated the minimal number of times.

Reply:
`Redis.sadd_returns_boolean` is just an `attr_accessor`, so I don't think it's that important to cache the lookup.

Reply:
well, it was really the `respond_to?` that I wanted to avoid calling a lot, and I didn't want to introduce any meta programming since that wasn't a pattern I saw used to evaluate fewer times. On the other hand, enable/disable isn't really a hotspots, so `¯\_(ツ)_/¯`

Thanks for the attention!